server/routes.go

```diff
@@ -1451,6 +1452,38 @@ func (s *Server) PsHandler(c *gin.Context) {
 	c.JSON(http.StatusOK, api.ProcessResponse{Models: models})
 }
 
+func (s *Server) TemplateHandler(w http.ResponseWriter, r *http.Request) {
+	var req api.TemplateRequest
+	if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	model, err := GetModel(req.Model)
+	if err != nil {
+		switch {
+		case os.IsNotExist(err):
+			http.Error(w, fmt.Sprintf("model '%s' not found", req.Model), http.StatusNotFound)
+		case err.Error() == "invalid model name":
+			http.Error(w, err.Error(), http.StatusBadRequest)
+		default:
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+		}
+		return
+	}
+
+	prompt, err := applyTemplate(model, req.Messages, req.Tools)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if err := json.NewEncoder(w).Encode(api.TemplateResponse{TemplatedPrompt: prompt}); err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
```

Comment:
Dangerous. Avoid sending back whatever error message happens to land in our laps. It could contain things we do not want to expose to clients. Please just say "Internal Server Error.".

If this is fixable by the client because they are in error, which they may be because this could be they gave us invalid json, they say that and use 400, otherwise, just bomb out and log.. carefully.

See the json package for types to assert on.

Reply:
Yeah you're right - will wrap thanks